var/spack/repos/builtin/packages/nseg/package.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+from shutil import copyfile
+
+
+class Nseg(MakefilePackage):
+    """NSEG - Low complexity sequence identification"""
+
+    homepage = "ftp://ftp.ncbi.nih.gov/pub/seg/nseg/"
+    url      = "ftp://ftp.ncbi.nih.gov/pub/seg/nseg/nseg.c"
+
+    version('1.0', sha256='afae364fbe51caa1eccb704ed0b954437a3ef48a0cd3bf473f4b61d6fde9617e',
+            expand=False)
+
+    resource(
+        name='makefile',
+        url="ftp://ftp.ncbi.nih.gov/pub/seg/nseg/makefile",
+        sha256='32937aef6969550ca3c762b5dd61b1520635cc46e773e35e56c5718f75838cee',
+        expand=False,
+    )
+
+    resource(
+        name='genwin.c',
+        url="ftp://ftp.ncbi.nih.gov/pub/seg/nseg/genwin.c",
+        sha256='d392d2db625dc8c90b00f2a86028a3a45d121e15eb403b51c2f9b01692ab10d9',
+        expand=False,
+    )
+
+    resource(
+        name='genwin.h',
+        url="ftp://ftp.ncbi.nih.gov/pub/seg/nseg/genwin.h",
+        sha256='1c701d87bf6200bfa40faa16fe665828a010727ef1aa0e8a1e5823605165fb86',
+        expand=False,
+    )
+
+    resource(
+        name='lnfac.h',
+        url="ftp://ftp.ncbi.nih.gov/pub/seg/nseg/lnfac.h",
+        sha256='5048e4f3dc3a7ea420d4eb4912a661f285634fbb205411b647b1f00c3fe3a0d2',
+        expand=False,
+    )
+
+    resource(
+        name='nmerge.c',
+        url="ftp://ftp.ncbi.nih.gov/pub/seg/nseg/nmerge.c",
+        sha256='c8a4bb4c960acf7fcd7509b635766b618efdab9f09aec36443040759eca3bce3',
+        expand=False,
+    )
+
+    resource(
+        name='runnseg',
+        url="ftp://ftp.ncbi.nih.gov/pub/seg/nseg/runnseg",
+        sha256='2830a5a1c5ea1a879cf3a415dfbb23db7a81e84d41698ddf765f2e1ef42e7c78',
+        expand=False,
+    )
+
+    build_directory = 'nseg'
+
+    @run_before('build')
+    def prepare_source(self):
+        # create the build dir
+        mkdirp(self.build_directory)
+
+        # move the primary source file in
+        copyfile('nseg.c', join_path(self.build_directory, 'nseg.c'))
+
+        # move all of the single-file resources into the build dir
+        for key in self.resources:
+            for res in self.resources[key]:
+                res_name = res.name
+                res_path = join_path(res.fetcher.stage.path, res.name)
+                copyfile(res_path, join_path(self.build_directory, res_name))
```

Comment:
You can use `copy` instead of `shutil.copyfile`. It's defined [here](https://spack.readthedocs.io/en/latest/llnl.util.html#llnl.util.filesystem.copy) and automatically imported for you.

But doesn't `resource` have some kind of destination argument that lets you tell Spack where to copy it automatically?

Reply:
When fetching the single-file resources spack places them in a subdirectory instead of directly at the destination. The directory name included a hash so I grabbed it from the resource object.

Updated to use `copy`.

Reply:
@codeandkey 

Ideally you could specify the `destination` like:

```
    resource(
        name='genwin.c',
        url="ftp://ftp.ncbi.nih.gov/pub/seg/nseg/genwin.c",
        sha256='d392d2db625dc8c90b00f2a86028a3a45d121e15eb403b51c2f9b01692ab10d9',
        expand=False,
        destination='nseg'
    )
```

and `genwin.c` would end up there automatically without having to copy them automatically. If you have a chance to try this out tomorrow and it works, that's great. If you don't (or it goes wrong - I have some pending fixes to resource management) then I'm fine merging this as is (since ideally we can make this whole issue go away in the future by implementing an alternate fetcher).

Reply:
Using `nseg` as the destination produces the following structure: `(stage/nseg-1.0-...)/nseg/resource-genwin.c-ai4iqipv4fqfux63b74zpioycrss5qs5/genwin.c` -- I thought this was (reasonably) unpredictable so the workaround dynamically queries the resource location.